src/magic.cpp

```diff
@@ -348,6 +346,21 @@ void spell_type::load( const JsonObject &jo, const std::string & )
     for( const JsonMember member : jo.get_object( "learn_spells" ) ) {
         learn_spells.insert( std::pair<std::string, int>( member.name(), member.get_int() ) );
     }
+
+    if( jo.has_member( "effect_str" ) ) {
+        if( jo.has_string( "effect_str" ) ) {
+            effect_str.emplace_back( jo.get_string( "effect_str" ) );
+        } else if( effect_name == "effect_on_condition" ) {
+            for( JsonValue jv : jo.get_array( "effect_str" ) ) {
+                effect_str.emplace_back( effect_on_conditions::load_inline_eoc( jv, "" ).c_str() );
+            }
+        } else {
+            mandatory( jo, was_loaded, "effect_str", effect_str );
+        }
+    }
+    if( effect_str.empty() ) {
+        effect_str.push_back( effect_str_default );
+    }
```

Comment:
i don't want the array in here. it's one-off code, it makes the loading code more messy, and you already mentioned that you can do effect_on_condition with a single string by utilizing nested conditions.